packages/core/src/shared/sam/build.ts

```diff
@@ -229,23 +235,35 @@ export async function runBuild(arg?: TreeNode): Promise<SamBuildResult> {
     await updateRecentResponse(buildMementoRootKey, 'global', 'templatePath', templatePath)
 
     try {
-        const { path: samCliPath } = await getSamCliPathAndVersion()
+        await vscode.window.withProgress(
+            {
+                location: vscode.ProgressLocation.Notification,
+            },
+            async (progress) => {
+                progress.report({ message: `Building SAM template at ${params.template.uri.path}` })
```

Comment:
can we also add a cancel button to it? you can refer the implementation here: 
https://github.com/aws/aws-toolkit-vscode/blob/5bfb867831cb9e8c475de5fb062607f37aa2991b/packages/core/src/shared/utilities/cliUtils.ts#L235-L238